usaspending_api/disaster/v2/views/award/count.py

```diff
@@ -0,0 +1,40 @@
+from django.db.models import Count
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from usaspending_api.common.cache_decorator import cache_response
+from usaspending_api.disaster.v2.views.count_base import CountBase
+from usaspending_api.awards.models import FinancialAccountsByAwards
+from usaspending_api.disaster.v2.views.disaster_base import FabaOutlayMixin, AwardTypeMixin
+
+
+class AwardCountViewSet(CountBase, FabaOutlayMixin, AwardTypeMixin):
+    """
+    Obtain the count of Agencies related to supplied DEFC filter.
+    """
+
+    endpoint_doc = "usaspending_api/api_contracts/contracts/v2/disaster/award/count.md"
+
+    required_filters = ["def_codes", "award_type_codes"]
+
+    @cache_response()
+    def post(self, request: Request) -> Response:
+
+        filters = [
+            self.is_in_provided_def_codes(),
+            self.all_closed_defc_submissions,
+            self.is_non_zero_award_cpe(),
+        ]
+        if self.award_type_codes:
+            count = (
+                FinancialAccountsByAwards.objects.filter(*filters)
+                .values("award_id")
+                .filter(award__type__in=self.filters.get("award_type_codes"))
+                .count()
```

Comment:
This count needs to be distinct as well since FABA -> awards is many to one

Reply:
Is this logic supposed to be unique on award id? The logic used by the amount endpoint (here transposed to https://github.com/fedspendingtransparency/usaspending-api/pull/2569/files#diff-5a427bd0e842bfcd87ac12ab1e696f58R154) uses four different FABA values.

Reply:
The four fields in FABA are used in an attempt to determine uniqueness when not joining to the awards table. When joining to the awards table, I assume counting the unique FKs would be more performant